src/Client.Infrastructure/Authentication/AzureAd/AzureAdAuthenticationService.cs

```diff
@@ -14,22 +14,16 @@ internal class AzureAdAuthenticationService : IAuthenticationService
 
     public AuthProvider ProviderType => AuthProvider.AzureAd;
 
-    public Task<IResult> Login(TokenRequest model) =>
+    public Task<IResult> LoginAsync(TokenRequest model) =>
         throw new NotImplementedException();
 
-    public async Task<IResult> Logout()
+    public async Task<IResult> LogoutAsync()
     {
         await _signOut.SetSignOutState();
         _navigation.NavigateTo("authentication/logout");
         return await Result.SuccessAsync();
     }
 
-    public Task<string> RefreshToken() =>
-        throw new NotImplementedException();
-
-    public Task<string> TryForceRefreshToken() =>
-        throw new NotImplementedException();
-
-    public Task<string> TryRefreshToken() =>
+    public Task<IResult<TokenResponse>> RefreshTokenAsync(RefreshTokenRequest request) =>
```

Comment:
refreshing is not needed here?

Reply:
Nope, this should be handled by AzureAdAuthorizationMessageHandler.

Reply:
The only reason this interface is implemented here is for the "LogOutAsync" method which gets called from some other places (and which is different for azuread then for jwt). Maybe we could split that interface up into 2 interfaces, so azuread only needs to implement one of them?